src/model/openthaigpt_pretraining_model/data_wrapper.py

```diff
@@ -1,37 +1,176 @@
 import torch
 from torch.utils.data import IterableDataset
+from datasets import Dataset, load_dataset, load_from_disk
+import os
+from re import findall
+from tqdm import tqdm
+from transformers import (
+    AutoTokenizer,
+    LlamaTokenizer,
+)
+from ..datasets.constants import SPLIT_TRAIN, SPLIT_VAL
 
-HF_TOKENIZER_INPUT_IDS_NAME = "input_ids"
 
+class TokenizedDataset:
+    def __init__(
+        self,
+        mode: str,
+        tokenizer: str,
+        max_tokens: int = 2048,
+        save_path: str = "./",
+        chunk_size: int = 1024 * 1024,
+        batch_size: int = 10000,
+        num_proc: int = 16,
+        dataset_name: str = "oscar",
+        dataset_dir: str = "unshuffled_deduplicated_th",
+    ):
+        if len(findall("llama", tokenizer)):
```

Comment:
We should load hf dataset before using `TokenizedDataset`. We have an example at [here](https://github.com/OpenThaiGPT/openthaigpt-pretraining/blob/1e1c481f78be18552bb154fdf66d78e761560817/src/model/openthaigpt_pretraining_model/lightning/utils.py#L92).

Reply:
do you mean input to be datasets and tokenize?

Reply:
Yes

Reply:
We may have customized dataset or tokenizer we should handle these once.

Reply:
one is datasets already tokenized and the other is not.

Reply:
done